src/storage/src/sink/kafka.rs

```diff
@@ -793,18 +788,16 @@ impl KafkaSinkState {
             let mut latest_ts = None;
             let mut latest_offset = None;
 
+            let progress_key_bytes = progress_key.as_bytes();
             while let Some((key, message, offset)) = get_next_message(&mut consumer, timeout)? {
                 debug_assert!(offset >= latest_offset.unwrap_or(0));
                 latest_offset = Some(offset);
 
-                let timestamp_opt = {
-                    match std::str::from_utf8(&key) {
-                        Ok(str_key) if str_key == consistency_key => {
-                            let progress: ProgressRecord = serde_json::from_slice(&message)?;
-                            Some(progress.timestamp)
-                        }
-                        _ => None,
-                    }
+                let timestamp_opt = if &key == progress_key_bytes {
```

Comment:
I avoided the problem here by comparing bytes; also has the benefit of being slightly faster, though unlikely we care here.

Reply:
nice, thanks!